### PR TITLE
[Az.Resources] add cmdlets for application federated identity credential

### PR DESCRIPTION
**Pull request by VeryEarly**
## Checklist

- [ ] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/Resources/MSGraph.Autorest/custom/New-AzADApplicationFederatedIdentityCredential.ps1 [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+{{ Add output here }}
+.Example
+PS C:\> {{ Add code here }}
+
+{{ Add output here }}
+
+.Outputs
+Microsoft.Azure.PowerShell.Cmdlets.Resources.MSGraph.Models.ApiV10Beta.IMicrosoftGraphFederatedIdentityCredential
+.Link
+https://docs.microsoft.com/powershell/module/az.resources/new-azadapplicationfederatedidentitycredential
```
Comment (wyunchi-ms) [inline src/Resources/MSGraph.Autorest/custom/New-AzADApplicationFederatedIdentityCredential.ps1:34]: We can remove these fields in custom file

**Review by wyunchi-ms — DISMISSED** [review]

File: src/Resources/MSGraph.Autorest/Az.MSGraph.psd1 [hunk 2]
```diff
@@ -11,7 +11,7 @@
...
+  FunctionsToExport = 'Add-AzADAppPermission', 'Add-AzADGroupMember', 'Get-AzADAppCredential', 'Get-AzADApplication', 'Get-AzADApplicationFederatedIdentityCredential', 'Get-AzADAppPermission', 'Get-AzADGroup', 'Get-AzADGroupMember', 'Get-AzADServicePrincipal', 'Get-AzADSpCredential', 'Get-AzADUser', 'New-AzADAppCredential', 'New-AzADApplication', 'New-AzADapplicationfederatedidentitycredential', 'New-AzADGroup', 'New-AzADServicePrincipal', 'New-AzADSpCredential', 'New-AzADUser', 'Remove-AzADAppCredential', 'Remove-AzADApplication', 'Remove-AzADapplicationfederatedidentitycredential', 'Remove-AzADAppPermission', 'Remove-AzADGroup', 'Remove-AzADGroupMember', 'Remove-AzADServicePrincipal', 'Remove-AzADSpCredential', 'Remove-AzADUser', 'Update-AzADApplication', 'Update-AzADapplicationfederatedidentitycredential', 'Update-AzADGroup', 'Update-AzADServicePrincipal', 'Update-AzADUser', '*'
```
Comment (dingmeng-xue) [inline src/Resources/MSGraph.Autorest/Az.MSGraph.psd1:14]: Some casing is incorrect.

File: src/Resources/MSGraph.Autorest/custom/New-AzADApplicationFederatedIdentityCredential.ps1 [hunk 3]
```diff
@@ -0,0 +1,141 @@
+
```
Comment (dingmeng-xue) [inline src/Resources/MSGraph.Autorest/custom/New-AzADApplicationFederatedIdentityCredential.ps1:1]: Why is this custom required?

File: src/Resources/MSGraph.Autorest/custom/Get-AzADApplicationFederatedIdentityCredential.ps1 [hunk 4]
```diff
@@ -0,0 +1,163 @@
+
+# ----------------------------------------------------------------------------------
```
Comment (dingmeng-xue) [inline src/Resources/MSGraph.Autorest/custom/Get-AzADApplicationFederatedIdentityCredential.ps1:2]: If Application is considered as parent object of FederatedIdentity, autorest doesn't support pipeline parameter object. We don't need this one for the time being. What's your thought?

File: src/Resources/MSGraph.Autorest/custom/New-AzADApplicationFederatedIdentityCredential.ps1 (hunk 3 above)
Comment (VeryEarly) [reply src/Resources/MSGraph.Autorest/custom/New-AzADApplicationFederatedIdentityCredential.ps1:1]: These parameters are mandatory but generated optional, I didn't find any way to convert them by directive

File: src/Resources/MSGraph.Autorest/README.md [hunk 5]
```diff
@@ -136,10 +137,43 @@ directive:
...
+
+  - where:
+      subject: ^applicationfederatedidentitycredentials$
+      verb: Get|New
+    hide: true
+
+  - where:
+      subject: ^applicationfederatedidentitycredentials$
+    set: 
+      subject: applicationfederatedidentitycredential
```
Comment (dingmeng-xue) [inline src/Resources/MSGraph.Autorest/README.md:170]: Can we change cmdlet name into {}-AzAdAppFederatedIdentityCredential?

File: src/Resources/MSGraph.Autorest/custom/Get-AzADApplicationFederatedIdentityCredential.ps1 (hunk 4 above)
Comment (VeryEarly) [reply src/Resources/MSGraph.Autorest/custom/Get-AzADApplicationFederatedIdentityCredential.ps1:2]: I'll say it's going to be very common that customer might want to pipe an application into federated identity cmdlets, we can remove these when autorest have them supported but I suggest keep them for now

File: src/Resources/MSGraph.Autorest/README.md (hunk 5 above)
Comment (VeryEarly) [reply src/Resources/MSGraph.Autorest/README.md:170]: > Can we change cmdlet name into {}-AzAdAppFederatedIdentityCredential?

shall I change these names in next release and link the current names as alias?

File: src/Resources/MSGraph.Autorest/custom/Get-AzADApplicationFederatedIdentityCredential.ps1 (hunk 4 above)
Comment (dingmeng-xue) [reply src/Resources/MSGraph.Autorest/custom/Get-AzADApplicationFederatedIdentityCredential.ps1:2]: OK

File: src/Resources/MSGraph.Autorest/README.md (hunk 5 above)
Comment (dingmeng-xue) [reply src/Resources/MSGraph.Autorest/README.md:170]: If possible, I hope we can use short name.

**Review by dingmeng-xue — APPROVED** [review]

**Pull request merged**